### PR TITLE
post to user activity log

### PR DESCRIPTION
**Pull request by jakect**
Note that they do not post to the main feed, that is determined by some custom algo. Apparentyly having custom content from the user makes it more likely that that will happen.

image

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -52,28 +59,53 @@ function renderMFS() {
...
+    // make connection post on behalf of user
+    // https://developers.facebook.com/docs/graph-api/reference/v2.3/user/feed#publish
+    var ogActionUri = '/me/' + appNs() + ':connect';
+    FB.api(ogActionUri, 'post',
+            {
+                message: "this is why I am connecting you people",
+                privacy: {
+                    'value': 'SELF'
+                },
+                profile: [], // TODO: pull in response.id from createProfileObjects
```
Comment (jzerbe) [inline app.js]: collection `sendUIDs` are merely tagging tokens. cannot be used here. fill out a bunch of [profile](https://developers.facebook.com/docs/reference/opengraph/object-type/profile#properties) entities for user here. need to map/reduce to store off all of the ids.

File: app.js [hunk 2]
```diff
@@ -52,28 +59,53 @@ function renderMFS() {
...
+                    'og:image': taggableFriend.picture.data.url,
+                    'profile:first_name': taggableFriend.first_name,
+                    'profile:last_name': taggableFriend.last_name,
+                    'profile:username': taggableFriend.name
+                },
+                privacy: {
+                    'value': 'SELF'
+                }
+            },
+            log
```
Comment (jzerbe) [inline app.js]: TODO: change this function to store off all returned `response.id`

File: app.js [hunk 3]
```diff
@@ -52,28 +59,53 @@ function renderMFS() {
...
+    // https://developers.facebook.com/docs/graph-api/reference/v2.3/user/feed#publish
+    var ogActionUri = '/me/' + appNs() + ':connect';
+    FB.api(ogActionUri, 'post',
+            {
+                message: "this is why I am connecting you people",
+                privacy: {
+                    'value': 'SELF'
+                },
+                profile: [], // TODO: pull in response.id from createProfileObjects
+                tags: sendUIDs.toString() // tagged users CSV
```
Comment (jzerbe) [inline app.js]: haven't been able to create action/profile(s) tuple to see how privacy `SELF` and tagging other users works

**Comment by jakect** [comment]
+1 nice improvements, merge when you're done.

**Comment by jzerbe** [comment]
@jakect rekt up the place good. my thoughts on what needs doing:

1. create a bunch of profile objects and store their ids
2. post action once all objects have been created

do these before merge?

**Comment by jakect** [comment]
Yeah, we can leave this open for now. I assume we'd just do that in a loop for now and concat the repsonses?

**Comment by jzerbe** [comment]
something like that. poor man's aync would be to fire to action posts that makes sure `sendUIDs.length() == ` array being passed to `profile` and doesn't execute on post until then

**Comment by jzerbe** [comment]
@jakect should be ready for review and merge

File: app.js [hunk 4]
```diff
@@ -52,28 +91,89 @@ function renderMFS() {
...
+    }
+
+    // make connection post on behalf of user
+    // https://developers.facebook.com/docs/graph-api/reference/v2.3/user/feed#publish
+    var ogActionUri = '/me/' + appNs() + ':connect';
+    FB.api(ogActionUri, 'post',
+            {
+                message: "this is why I am connecting you people",
+                privacy: {
+                    'value': 'SELF' // will be exposed to SELF and anyone tagged
```
Comment (jakect) [inline app.js]: :+1: good to know it automatically knows to do that.

**Comment by jakect** [comment]
+1, few minor followup items that can be looked into. We should sync up on what we want to do next.

**Pull request merged**